document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/accounts/AccountsManager.java

```diff
@@ -299,6 +300,45 @@ public SmallFullAccountIxbrl getSmallFullAccounts(String link, Transaction trans
                 smallFullApiData.setOffBalanceSheet(offBalanceSheetApi);
             }
 
+            if (!StringUtils.isEmpty(smallFull.getLinks().getLoansToDirectors())) {
+
+                errorString = "loans to directors";
+
+                LoansToDirectorsApi loansToDirectors =
+                        apiClient.smallFull().loansToDirectors()
+                                .get(smallFull.getLinks().getLoansToDirectors()).execute().getData();
+
+                smallFullApiData.setLoansToDirectors(loansToDirectors);
+
+                if (!loansToDirectors.getLoans().isEmpty()) {
+
+                    errorString = "loans";
+
+                    // Find the first loan id from the map of loans
+                    String loanId =
+                            loansToDirectors.getLoans()
+                                    .keySet().stream().findFirst()
+                                    .orElseThrow(RuntimeException::new);
```

Comment:
Throw an empty RuntimeException? Perhaps a more specific exception (would this be illegalState?), with a message?

Reply:
e.g.
`.orElseThrow(() -> new IllegalStateException("Loans to Directors link exists, but no loans found));`